src/webhook/job/manager/manager.go

```diff
@@ -0,0 +1,67 @@
+package manager
+
+import (
+	"fmt"
+
+	"github.com/goharbor/harbor/src/common/dao"
+	"github.com/goharbor/harbor/src/common/models"
+	"github.com/goharbor/harbor/src/webhook/job"
+)
+
+// DefaultManager ..
+type DefaultManager struct {
+}
+
+// NewDefaultManager ...
+func NewDefaultManager() job.Manager {
+	return &DefaultManager{}
+}
+
+// Create ...
+func (d *DefaultManager) Create(job *models.WebhookJob) (int64, error) {
+	return dao.AddWebhookJob(job)
+}
+
+// List ...
+func (d *DefaultManager) List(query ...*models.WebhookJobQuery) (int64, []*models.WebhookJob, error) {
+	total, err := dao.GetTotalCountOfWebhookJobs(query...)
+	if err != nil {
+		return 0, nil, err
+	}
+
+	executions, err := dao.GetWebhookJobs(query...)
+	if err != nil {
+		return 0, nil, err
+	}
+	return total, executions, nil
+}
+
+// Update ...
+func (d *DefaultManager) Update(job *models.WebhookJob, props ...string) error {
+	n, err := dao.UpdateWebhookJob(job, props...)
+	if err != nil {
+		return err
+	}
+
+	if n == 0 {
+		return fmt.Errorf("execution %d not found", job.ID)
+	}
+	return nil
+}
+
+// UpdateJobStatus ...
+func (d *DefaultManager) UpdateJobStatus(jobID int64, status string, statusCondition ...string) error {
```

Comment:
Could you document the  usage `statusCondition` and what will happen if the statusCondition does not match?

Reply:
> Could you document the usage `statusCondition` and what will happen if the statusCondition does not match?

`statusCondition` used to filter the jobs with status. 
the status of a job will be in 2 situations:
1. pending(init status)-> running(submit to jobservice)-> finished/error. we should change the status to `running` only when current status is `pending`. if the status is not `pending`, means jobservice already changed status to `finished` or `error`, will do nothing with the job.
2. pending(init status)-> error(submit failed).